diff --git a/demo/src/bugs/DelayedRender.js b/demo/src/bugs/DelayedRender.js
index da2802c..5f847ad 100644
--- a/demo/src/bugs/DelayedRender.js
+++ b/demo/src/bugs/DelayedRender.js
@@ -86,11 +86,7 @@ function formatAsCurrency(amount) {
 }
 
 function parseCurrencyAsAmount(currency) {
-  let amount = Number(currency.replace(/[^0-9.-]+/g, "")) * 100;
-
-  for (let i = 0; i < Math.pow(amount, 2.8); i++) {
-      Math.sqrt(i);
-  }
+  const amount = Number(currency.replace(/[^0-9.-]+/g, "")) * 100;
 
   return amount;
 }
